crates/uv-torch/src/accelerator.rs

```diff
@@ -0,0 +1,136 @@
+        }
+    }
+}
+
+impl Accelerator {
+    /// Detect the CUDA driver version from the system.
+    ///
+    /// Query, in order:
+    /// 1. The `UV_CUDA_DRIVER_VERSION` environment variable.
+    /// 2. `/sys/module/nvidia/version`, which contains the driver version (e.g., `550.144.03`).
+    /// 3. `/proc/driver/nvidia/version`, which contains the driver version among other information.
+    /// 4. `nvidia-smi --query-gpu=driver_version --format=csv,noheader`.
+    pub fn detect() -> Result<Option<Self>, AcceleratorError> {
+        // Read from `UV_CUDA_DRIVER_VERSION`.
+        if let Ok(driver_version) = std::env::var(EnvVars::UV_CUDA_DRIVER_VERSION) {
+            let driver_version = Version::from_str(&driver_version)?;
+            debug!("Detected CUDA driver version from `UV_CUDA_DRIVER_VERSION`: {driver_version}");
+            return Ok(Some(Self::Cuda { driver_version }));
+        }
+
+        // Read from `/sys/module/nvidia/version`.
+        match fs_err::read_to_string("/sys/module/nvidia/version") {
+            Ok(content) => {
+                return match parse_sys_module_nvidia_version(&content) {
+                    Ok(driver_version) => {
+                        debug!("Detected CUDA driver version from `/sys/module/nvidia/version`: {driver_version}");
+                        Ok(Some(Self::Cuda { driver_version }))
+                    }
+                    Err(e) => Err(e),
+                }
+            }
+            Err(e) if e.kind() == std::io::ErrorKind::NotFound => {}
+            Err(e) => return Err(e.into()),
+        }
+
+        // Read from `/proc/driver/nvidia/version`
+        match fs_err::read_to_string("/proc/driver/nvidia/version") {
+            Ok(content) => {
+                match parse_proc_driver_nvidia_version(&content) {
+                    Ok(Some(driver_version)) => {
+                        debug!("Detected CUDA driver version from `/proc/driver/nvidia/version`: {driver_version}");
+                        return Ok(Some(Self::Cuda { driver_version }));
+                    }
+                    Ok(None) => {
+                        debug!("Failed to parse CUDA driver version from `/proc/driver/nvidia/version`");
+                    }
+                    Err(e) => return Err(e),
+                }
+            }
+            Err(e) if e.kind() == std::io::ErrorKind::NotFound => {}
+            Err(e) => return Err(e.into()),
+        }
+
+        // Query `nvidia-smi`.
+        if let Ok(output) = std::process::Command::new("nvidia-smi")
+            .arg("--query-gpu=driver_version")
+            .arg("--format=csv,noheader")
+            .output()
+        {
+            if output.status.success() {
+                let driver_version = Version::from_str(&String::from_utf8(output.stdout)?)?;
+                debug!("Detected CUDA driver version from `nvidia-smi`: {driver_version}");
+                return Ok(Some(Self::Cuda { driver_version }));
+            }
```

Comment:
`else {debug!("nvidia-smi returned error {output.status}: {output.stderr}")}` might be nice